File: eu-1/first.js
'use strict'

// solution to problem 1

module.exports = function(limit){

  var multiples = []
  , sum = 0;

  function retain(num){
    if(multiples.indexOf(num) === -1) multiples.push(num);
  }

  for(var k = 0; k < limit; k++){
    if ( k % 3 === 0 ) retain(k);
    else if ( k % 5 === 0 ) retain(k);
  }

  sum = multiples.reduce(function(a, b){
    return a + b;
  });

  return sum;

};

File: 1-Multiples_of_3_and_5/first.c
#include <stdio.h>

int main(int argc, char *argv[])
{
  int limit = 1000;
  int k;
  long sum;

  sum = 0;

  for(k = 0; k < limit; k++){
    if(k % 3 == 0 || k % 5 == 0) sum = sum + k;
  }

  printf("Sum: %ld \n", sum);
    
  return 0;
}
